app/api/v1/users.rb

```diff
@@ -0,0 +1,19 @@
+class V1::Users < Grape::API
+  
+  before do
+    authenticate!
+    
+    @allowed_users = current_team&.users || User.all
```

Comment:
In case there is no current_team (which should be a bug) the api must return zero data back, not everything.